docs/method.md

```diff
@@ -0,0 +1,128 @@
+
+Unlike the example shown above, where we assume a constant value of 1.0 for the learnable parameters, in a supervised learning environment, the optimization algorithm will find a set of the best learnable parameters ($\boldsymbol{\Theta}_\phi, \boldsymbol{\Theta}_\gamma$) in the message passing operation.
+
+## Graph Neural Network-based Simulator (GNS)
+
+In this study, we use GNN as a surrogate simulator to model granular flow behavior. Figure 3 shows an overview of the general concepts and structure of the GNN-based simulator (GNS). Consider a granular flow domain represented as particles (Figure 3a). In GNS, we represent the physical state of the granular domain at time t with a set of $\boldsymbol{x}_i^t$ describing the state and properties of each particle. The GNS takes the current state of the granular flow $\boldsymbol{x}_t^i\in\boldsymbol{X}_t$ and predicts its next state ${\boldsymbol{x}_{i+1}^i\in\boldsymbol{X}}_{t+1}$ (Figure 3a). The GNS consists of two components: a parameterized function approximator $\ d_\mathbf{\Theta}$ and an updater function (Figure 3b). The approximator $d_\theta$ take takes $\boldsymbol{X}_t$ as an input and outputs dynamics information ${\boldsymbol{y}_i^t\in\boldsymbol{Y}}_t$. The updater then computes $\boldsymbol{X}_{t+1}$ using $\boldsymbol{Y}_t$ and $\boldsymbol{X}_t$. Figure 3c shows the details of $d_\theta$ which consists of an encoder, a processor, and a decoder. The encoder (Figure 3c-1) takes the state of the system $\boldsymbol{X}^t$ and embed it into a latent graph $G_0=\left(\boldsymbol{V}_0,\ \boldsymbol{E}_0\right)$ to represent the relationship between particles, where the vertices $\boldsymbol{v}_i^t \in\boldsymbol{V}_0$ contain latent information of the current particle state, and the edges $\boldsymbol{e}_{i,j}^t \in\boldsymbol{E}_0$ contain latent information of the pair-wise relationship between particles. Next, the processer (Figure 3c-2) converts $G_0$ to $G_M$ with $M$ stacks of message passing GNN ($G_0\rightarrow\ G_1\rightarrow\cdots\rightarrow\ G_M$)  to compute the interaction between particles. Finally, the decoder (Figure 3c-3) extracts dynamics of the particles ($\boldsymbol{Y}^t$) from $G_M$, such as the acceleration of the physical system. The entire simulation (Figure 3a) involves running GNS surrogate model through $K$ timesteps predicting from the initial state $\boldsymbol{X}_0$ to $\boldsymbol{X}_K$ $(\boldsymbol{X}_0,\ \ \boldsymbol{X}_1,\ \ \ldots,\ \ \boldsymbol{X}_K$), updating at each step ($\boldsymbol{X}_t\rightarrow\boldsymbol{X}_{t+1}$)
+
+![GNS](img/fig3.svg)
```

Comment:
Use names which are meaningful. Never name files as fig3.svg - it has no meaning and cannot be searched. Write one sentence per line so it is easier to diff.